dumper/app/src/main/java/com/google/edwmigration/dumper/application/dumper/connector/greenplum/AbstractGreenplumConnector.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright 2022-2024 Google LLC
+ * Copyright 2013-2021 CompilerWorks
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.edwmigration.dumper.application.dumper.connector.greenplum;
+
+import java.time.ZoneOffset;
+import java.time.format.DateTimeFormatter;
+import java.util.List;
+
+import javax.annotation.Nonnull;
+
+import com.google.common.base.Joiner;
+import com.google.edwmigration.dumper.application.dumper.annotations.RespectsArgumentDriver;
+import com.google.edwmigration.dumper.application.dumper.annotations.RespectsArgumentDriverClass;
+import com.google.edwmigration.dumper.application.dumper.annotations.RespectsArgumentHostUnlessUrl;
+import com.google.edwmigration.dumper.application.dumper.annotations.RespectsArgumentPassword;
+import com.google.edwmigration.dumper.application.dumper.annotations.RespectsArgumentUri;
+import com.google.edwmigration.dumper.application.dumper.annotations.RespectsArgumentUser;
+import com.google.edwmigration.dumper.application.dumper.connector.postgresql.AbstractPostgresqlConnector;
+
+/** @author zzwang */
+@RespectsArgumentDriver
+@RespectsArgumentDriverClass
+@RespectsArgumentHostUnlessUrl
+@RespectsArgumentUser
+@RespectsArgumentPassword
+@RespectsArgumentUri
+public abstract class AbstractGreenplumConnector extends AbstractPostgresqlConnector {
+
+    protected static final DateTimeFormatter SQL_FORMAT = DateTimeFormatter.ISO_OFFSET_DATE_TIME
+            .withZone(ZoneOffset.UTC);
+
+    public AbstractGreenplumConnector(String name) {
+        super(name);
+    }
+
+    @Nonnull
+    protected static CharSequence newWhereClause(List<String> clauseList, String... clauseArray) {
+        StringBuilder buf = new StringBuilder();
+        Joiner.on(" AND ").appendTo(buf, clauseList);
+        for (String clause : clauseArray) {
```

Comment:
It seems to me that `Joiner` and the for loop append the clauses twice. I would suggest removing one of them.

Reply:
So I copied this over from redshift and It actually wasn't appending it twice because, since clauseList and clauseArray are "different" sources, but since greenplum doesn't need clauseList I removed it.